cli/fathom_web/test/test_train.py

```diff
@@ -32,3 +33,13 @@ def test_auto_vectorization_smoke(tmp_path):
     )
     assert result.exit_code == 0
     assert (tmp_path / 'training_vectors.json').exists()
+
+
+def test_find_optimal_cutoff():
```

Comment:
I guess with all those corner cases that popped up above, we'll need some more tests! :-)